src/domain/user/hooks/useUserOrganizations.tsx

```diff
@@ -0,0 +1,73 @@
+import { useApolloClient } from '@apollo/client';
+import sortBy from 'lodash/sortBy';
+import uniq from 'lodash/uniq';
+import React from 'react';
+
+import {
+  OrganizationFieldsFragment,
+  UserFieldsFragment,
+} from '../../../generated/graphql';
+import useIsMounted from '../../../hooks/useIsMounted';
+import { getOrganizationQueryResult } from '../../organization/utils';
+
+type userOrganizationsState = {
+  loading: boolean;
+  organizations: OrganizationFieldsFragment[];
+};
+
+const useUserOrganizations = (
+  user?: UserFieldsFragment
+): userOrganizationsState => {
+  const isMounted = useIsMounted();
+  const apolloClient = useApolloClient();
+  const [loading, setLoading] = React.useState(false);
+  const [organizations, setOrganizations] = React.useState<
+    OrganizationFieldsFragment[]
+  >([]);
+
+  React.useEffect(() => {
+    const organizationIds = uniq([
+      user?.organization,
+      ...(user?.adminOrganizations ?? []),
+      ...(user?.organizationMemberships ?? []),
+    ]).filter((id) => id);
+
+    const getUserOrganizations = async () => {
+      try {
+        setLoading(true);
+
+        const userOrganizations = await Promise.all(
+          organizationIds.map(async (id) => {
+            const userOrganization = await getOrganizationQueryResult(
+              id as string,
+              apolloClient
+            );
+            return userOrganization;
+          })
+        );
```

Comment:
If I understand this correctly, this could be simplified little bit to be more concise:

```
 const userOrganizations = await Promise.all(
          organizationIds.map((id) =>
            getOrganizationQueryResult(id as string, apolloClient)
          )
        );
```

Reply:
Changes as suggested